panml/core/llm/huggingface.py

```diff
@@ -106,7 +112,10 @@ def predict(self, text: str, max_length: int=50, skip_special_tokens: bool=True,
             'perplexity': None,
         }
         
-        input_ids = self.tokenizer.encode(text, return_tensors='pt')
+        if 'cuda' in self.device:
+            input_ids = self.tokenizer.encode(text, return_tensors='pt').to(self.device)
```

Comment:
i think `.to(self.device)` just puts the data or model on device right? In that case i don't think you need an if condition for cleaner code. It can either `.to('cpu')` (which should be the same as not using `.to()` as the default should be cpu) or `.to('cuda')` depending on the device.

Reply:
Yeah I think it should be torch.device('cuda')

Reply:
Ok @Hanshan1988 yes I'll remove the if statement and just leave it as .to(torch.device(self.device))